x/blob/ante/blob_share_decorator_test.go

```diff
@@ -24,141 +30,138 @@ const (
+			name:        "PFB with 1 blob that is 2 MiB",
+			blobsPerPFB: 1,
+			blobSize:    2 * mebibyte,
+			appVersion:  v2.Version,
 			// This test case should return an error because a square size of 64
 			// has exactly 2 MiB of total capacity so the total blob capacity
 			// will be slightly smaller than 2 MiB.
 			wantErr: blob.ErrBlobsTooLarge,
 		},
 		{
-			name: "PFB with 2 blobs that are 1 byte each",
-			pfb: &blob.MsgPayForBlobs{
-				BlobSizes: []uint32{1, 1},
-			},
-			appVersion: v2.Version,
+			name:        "PFB with 2 blobs that are 1 byte each",
+			blobsPerPFB: 2,
+			blobSize:    1,
+			appVersion:  v2.Version,
 		},
 		{
-			name: "PFB with 2 blobs that are 1 MiB each",
-			pfb: &blob.MsgPayForBlobs{
-				BlobSizes: []uint32{mebibyte, mebibyte},
-			},
-			appVersion: v2.Version,
+			name:        "PFB with 2 blobs that are 1 MiB each",
+			blobsPerPFB: 2,
+			blobSize:    1 * mebibyte,
+			appVersion:  v2.Version,
 			// This test case should return an error for the same reason a
 			// single blob that is 2 MiB returns an error.
 			wantErr: blob.ErrBlobsTooLarge,
 		},
 		{
-			name: "PFB with many single byte blobs should fit",
-			pfb: &blob.MsgPayForBlobs{
-				// 4095 blobs each of size 1 byte should occupy 4095 shares.
-				// When square size is 64, there are 4095 shares available to
-				// blob shares so we don't expect an error for this test case.
-				BlobSizes: repeat(4095, 1),
-			},
-			appVersion: v2.Version,
+			name:        "PFB with many single byte blobs should fit",
+			blobsPerPFB: 3000,
+			blobSize:    1,
+			appVersion:  v2.Version,
 		},
 		{
-			name: "PFB with too many single byte blobs should not fit",
-			pfb: &blob.MsgPayForBlobs{
-				// 4096 blobs each of size 1 byte should occupy 4096 shares.
-				// When square size is 64, there are 4095 shares available to
-				// blob shares so we expect an error for this test case.
-				BlobSizes: repeat(4096, 1),
-			},
-			appVersion: v2.Version,
-			wantErr:    blob.ErrBlobsTooLarge,
+			name:        "PFB with too many single byte blobs should not fit",
+			blobsPerPFB: 4000,
+			blobSize:    1,
+			appVersion:  v2.Version,
+			wantErr:     blob.ErrBlobsTooLarge,
 		},
 		{
-			name: "PFB with 1 blob that is 1 share",
-			pfb: &blob.MsgPayForBlobs{
-				BlobSizes: []uint32{uint32(share.AvailableBytesFromSparseShares(1))},
-			},
-			appVersion: v2.Version,
+			name:        "PFB with 1 blob that is 1 share",
+			blobsPerPFB: 1,
+			blobSize:    100,
+			appVersion:  v2.Version,
 		},
 		{
-			name: "PFB with 1 blob that occupies total square - 1",
-			pfb: &blob.MsgPayForBlobs{
-				BlobSizes: []uint32{uint32(share.AvailableBytesFromSparseShares((squareSize * squareSize) - 1))},
-			},
-			appVersion: v2.Version,
+			name:        "PFB with 1 blob that occupies total square - 1",
+			blobsPerPFB: 1,
+			blobSize:    share.AvailableBytesFromSparseShares(squareSize*squareSize - 1),
+			appVersion:  v2.Version,
+			wantErr:     blob.ErrBlobsTooLarge,
```

Comment:
[question] why does this error now? If the blob occupies the total square - 1 then there should be room for 1 share to contain the PFB tx.

Reply:
hmm that's a good question, let me check

Reply:
good catch!! 45251c1757086723335e2625fdc48af9cd4bb4ad